test/unit/test_diagram.py

```diff
@@ -46,6 +48,32 @@ def test_job_agg_data():
     assert job_agg.number_of_columns() == 6
 
 
+def test_seq():
+    """
+    function to test the Class JobData inside autosubmit/monitor/diagram.py
+    """
+    for value in diagram._seq(100, 2, 10):
```

Comment:
Maybe test the `len` of the seq? In this case it must have `9` elements, `100, 110, 120, 130, 140, 150, 160, 170, 180`. This way, if `_seq` is broken and returns less/none elements, this test would fail (otherwise, the for would not enter, and the `assert`s would never be called -- so test wouldn't fail).